Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1085 @@
+
+    // Do the extensions in the x-plane
+    int idim = 0;
+
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+
+        amrex::Box const &box = mfi.validbox();
+
+        auto const &Sx = m_face_areas[maxLevel()][idim]->array(mfi);
+        auto const &flag_ext_face_x = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+        auto const &flag_info_face_x = m_flag_info_face[maxLevel()][idim]->array(mfi);
+        auto &borrowing_x = (*m_borrowing[maxLevel()][idim])[mfi];
+        auto const &borrowing_x_inds_pointer = borrowing_x.inds_pointer.array();
+        auto const &borrowing_x_size = borrowing_x.size.array();
+        amrex::Long ncells = box.numPts();
+        int* borrowing_x_inds = borrowing_x.inds.data();
+        uint8_t* borrowing_x_neigh_faces = borrowing_x.neigh_faces.data();
+        amrex::Real* borrowing_x_area = borrowing_x.area.data();
+
+        auto const &Sx_mod = m_area_mod[maxLevel()][idim]->array(mfi);
+        const auto &ly = m_edge_lengths[maxLevel()][1]->array(mfi);
+        const auto &lz = m_edge_lengths[maxLevel()][2]->array(mfi);
+        const amrex::Real dy = cell_size[1];
+        const amrex::Real dz = cell_size[2];
+
+        nelems_x = amrex::Scan::PrefixSum<int>(ncells,
+        [=] AMREX_GPU_DEVICE (int icell) {
+            const amrex::Dim3 cell = box.atOffset(icell).dim3();
+            const int i = cell.x;
+            const int j = cell.y;
+            const int k = cell.z;
+            // If the face doesn't need to be extended break the loop
+            if (!flag_ext_face_x(i, j, k)) {
+                return 0;
+            }
+
+            amrex::Real Sx_stab = 0.5 * std::max({ly(i, j, k) * dz, ly(i, j, k + 1) * dz,
+                                                  lz(i, j, k) * dy, lz(i, j + 1, k) * dy});
+            amrex::Real Sx_ext = Sx_stab - Sx(i, j, k);
+            int n_borrow = ComputeNBorrowOneFaceExtension(cell, Sx_ext, Sx_mod, flag_info_face_x,
+                                                          flag_ext_face_x, idim);
+
+            borrowing_x_size(i, j, k) = n_borrow;
+            return n_borrow;
+        },
+        [=] AMREX_GPU_DEVICE (int icell, int ps){
+            amrex::Dim3 cell = box.atOffset(icell).dim3();
+            const int i = cell.x;
+            const int j = cell.y;
+            const int k = cell.z;
+            int nborrow = borrowing_x_size(i, j, k);
+            if (nborrow == 0) {
+                borrowing_x_inds_pointer(i, j, k) = nullptr;
+            } else{
+                borrowing_x_inds_pointer(i, j, k) = borrowing_x_inds + ps;
+
+                amrex::Real Sx_stab = 0.5 * std::max({ly(i, j, k) * dz, ly(i, j, k + 1) * dz,
+                                                      lz(i, j, k) * dy, lz(i, j + 1, k) * dy});
+                amrex::Real Sx_ext = Sx_stab - Sx(i, j, k);
+                for (int j_n = -1; j_n < 2; j_n++) {
+                    for (int k_n = -1; k_n < 2; k_n++) {
```

Comment:
Sorry Remi, I overlooked this comment. Unfortunately, this can be an issue. 
Probably one way to correct it is to have a copy of Sx_mod that we use only in `ComputeNBorrowOneFaceExtension` to keep track of the area which has already been subtracted from a cell. Although this might work, it's not a great solution.